crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -505,6 +517,15 @@ impl<'a> Iterator for LinePreprocessor<'a> {
                     // Set the start for the next logical line.
                     self.line_start = range.end();
 
+                    if self
+                        .cell_offsets
+                        .is_some_and(|cell_offsets| cell_offsets.contains(&self.line_start))
```

Comment:
You may want to use `containing_range` here. `contains` is `O(n)`, `containing_range` is `O(log(n))`. But I wonder if we could do better by keeping an iterator with the cell offsets (they're sorted) and peek at the first offset and:

* while it is smaller than the `line_start`, call next 
* if it is not `None`, you know its inside of the cell. 
This gives you `O(n)` performance

Reply:
@MichaReiser It's not exactly what you said, but I gave it a try in 3f703afa6. What do you think ?